pkg/network/ebpf/c/tracer/maps.h

```diff
@@ -36,7 +36,7 @@ BPF_PERF_EVENT_ARRAY_MAP(conn_close_event, __u32)
  * or BPF_MAP_TYPE_PERCPU_ARRAY, but they are not available in
  * some of the Kernels we support (4.4 ~ 4.6)
  */
-BPF_HASH_MAP(conn_close_batch, __u32, batch_t, 1024)
+BPF_HASH_MAP(conn_close_batch, __u32, batch_t, 1)
```

Comment:
nit: I think typically we set map sizes to 0 when we intend to overwrite them in userspace

Reply:
That is for maps the must be resized. This can remain at `1` if it is not being used, but must be included because the code references it.

Reply:
makes sense 👍

Reply:
I didn't know we could remove maps from the spec at load time? If so it's likely a trivial difference in memory footprint but a good pattern nonetheless

Reply:
> I think setting this to 0 is still a good safeguard

Safeguard against what? The default configuration all matches at the moment. Changing this to `0` means, that you must resize the map, even if using the default value for whether or not to do the custom batching.

> ideally remove this from the map spec if not required, at load time

I don't think we can completely remove the map spec. This is because there is still code that references that map, even though it is protected by a branch that will never get taken.

Reply:
> Safeguard against what?    

Against loading the map with max entries set to 1, because the userspace forgot to resize it. This may happen during a refactor, when someone moves the code around. Having a default value of `0` forces the userspace to think about the correct value under all conditions.

Reply:
max entries set to 1 is the desired value when custom batching is disabled.

Reply:
If you forgot to resize, then the batch manager will fail loudly when it is trying to setup the default map values.